### PR TITLE
[Task 1852325] load SDK categories from static file

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1852325](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1852325)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public class AzureSdkCategoryService {
+    private static final ObjectMapper CSV_MAPPER = new CsvMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private static final String SERVICE_CATEGORY_CSV = "/service-category.csv";
+
+    @Cacheable(value = "azure-sdk-category-entities")
+    @AzureOperation(name = "sdk.load_meta_data", type = AzureOperation.Type.TASK)
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:34]: rename operation

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public static Map<String, List<AzureSdkCategoryEntity>> loadAzureSDKCategories() {
+        try (final InputStream stream = WorkspaceTaggingService.class.getResourceAsStream(SERVICE_CATEGORY_CSV)) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:36]: should not be `WorkspaceTaggingService`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            // read
+            final ObjectReader reader = CSV_MAPPER.readerFor(AzureSdkCategoryEntity.class).with(CsvSchema.emptySchema().withHeader());
+            final MappingIterator<AzureSdkCategoryEntity> data = reader.readValues(stream);
+            final List<AzureSdkCategoryEntity> categories = data.readAll();
+            // default category & description suffix.
+            categories.stream().filter(c -> StringUtils.isNotBlank(c.getServiceName())).forEach(c -> {
+                if (StringUtils.isBlank(c.getCategory())) {
+                    c.setCategory("Others");
+                }
+                if (StringUtils.isNotBlank(c.getDescription()) && !StringUtils.endsWith(c.getDescription(), ".")) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:46]: may need trim first

**Review by Flanker32 — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java (hunk 1 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:34]: renamed.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java (hunk 2 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:36]: updated in https://github.com/microsoft/azure-tools-for-java/pull/5400

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java (hunk 3 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkCategoryService.java:46]: updated in https://github.com/microsoft/azure-tools-for-java/pull/5400

**Pull request merged**